File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        
        int[] DP = new int[n+1];
        int ind = 0;
        return helper(DP, n, ind);
    }
    
    int helper(int[] DP, int n, int ind){
        if (ind == n) return 1;
        if (ind>n) return 0;
        if (DP[ind] != 0) return DP[ind];
        
        
        return DP[ind] = helper(DP, n, ind+1) + helper(DP, n, ind+2);
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [-1] * (n+1)
        
        def helper(index):
            if index == n:
                return 1
            if index > n:
                return 0
            if dp[index] != -1: return dp[index]
            
            dp[index] = helper(index + 1) + helper(index + 2)
            return dp[index]
    
        return helper(0)